src/Core/Parser/SyntaxTree/Expression/AbstractExpressionNode.php

```diff
@@ -52,7 +51,7 @@ public function __construct(string $expression, array $matches)
      * @param RenderingContextInterface $renderingContext
      * @return string the text stored in this node/subtree.
      */
-    public function evaluate(RenderingContextInterface $renderingContext)
+    public function evaluate(RenderingContextInterface $renderingContext): mixed
```

Comment:
Interesting ... phpdocblock says `@return string` ... that does not match ...

Reply:
The `CastingExpressionNode` can return various types, so `mixed` is correct here. I removed the phpdoc.